core/amber/src/main/scala/edu/uci/ics/texera/web/resource/WorkflowWebsocketResource.scala

```diff
@@ -11,38 +10,46 @@ import edu.uci.ics.amber.engine.common.tuple.ITuple
 import edu.uci.ics.texera.web.TexeraWebApplication
 import edu.uci.ics.texera.web.model.event._
 import edu.uci.ics.texera.web.model.request._
-import edu.uci.ics.texera.web.resource.WorkflowWebsocketResource.{sessionJobs, sessionResults}
+import edu.uci.ics.texera.web.resource.WorkflowWebsocketResource.sessionResults
+import edu.uci.ics.texera.web.resource.auth.UserResource
 import edu.uci.ics.texera.workflow.common.tuple.Tuple
 import edu.uci.ics.texera.workflow.common.workflow.{WorkflowCompiler, WorkflowInfo}
 import edu.uci.ics.texera.workflow.common.{Utils, WorkflowContext}
 import edu.uci.ics.texera.workflow.operators.sink.SimpleSinkOpDesc
 
-import javax.websocket._
+import java.util.concurrent.atomic.AtomicInteger
+import javax.servlet.http.HttpSession
 import javax.websocket.server.ServerEndpoint
+import javax.websocket.{EndpointConfig, _}
 import scala.collection.mutable
-
 object WorkflowWebsocketResource {
 
-  val nextWorkflowID = new AtomicInteger(0)
+  val nextJobID = new AtomicInteger(0)
 
   val sessionMap = new mutable.HashMap[String, Session]
   val sessionJobs = new mutable.HashMap[String, (WorkflowCompiler, ActorRef)]
   val sessionResults = new mutable.HashMap[String, Map[String, List[ITuple]]]
 }
 
-@ServerEndpoint("/wsapi/workflow-websocket")
+@ServerEndpoint(value = "/wsapi/workflow-websocket", configurator = classOf[ServletAwareConfig])
 class WorkflowWebsocketResource {
 
   final val objectMapper = Utils.objectMapper
 
+  private var config: EndpointConfig = _
+
   @OnOpen
-  def myOnOpen(session: Session): Unit = {
+  def myOnOpen(session: Session, config: EndpointConfig): Unit = {
     WorkflowWebsocketResource.sessionMap.update(session.getId, session)
+    this.config = config
```

Comment:
Do we need to associate the config object with each websocket session? Because if there are two concurrent users, the second onOpen will override the first config object. I checked that the config object is different for different onOpen calls.

Reply:
Technically speaking, it is not needed. 

`WorkflowWebsocketResource` is a `ServerEndpoint` which each websocket connection will be handled by a different thread. Each instance of `WorkflowWebsocketResource` will have their own copy of `EndpointConfig`. Thus won't be overwriting each other.

However, to make the code cleaner, I instead update the httpSession to the singleton (object class) `WorkflowWebsocketResource.sessionMap` along with websocketSession.